diff --git a/currency_rate_update/services/currency_getter_interface.py b/currency_rate_update/services/currency_getter_interface.py
index fc2b045..881360e 100644
--- a/currency_rate_update/services/currency_getter_interface.py
+++ b/currency_rate_update/services/currency_getter_interface.py
@@ -20,10 +20,13 @@
 ##############################################################################
 
 import logging
+import re
 
 from datetime import datetime
 from openerp import fields
 from openerp.exceptions import except_orm
+from bs4 import BeautifulSoup, Comment, NavigableString
+from selenium import webdriver
 
 _logger = logging.getLogger(__name__)
 
@@ -90,7 +93,7 @@ class Currency_getter_interface(object):
     ]
 
     # Updated currency this arry will contain the final result
-    updated_currency = {}
+    updated_currency = {'rate':{}, 'rate_buy':{}, 'rate_sell':{}, 'rate_statistics':{}}
 
     def get_updated_currency(self, currency_array, main_currency,
                              max_delta_days):